tests/core/pyspec/eth2spec/test/phase0/epoch_processing/test_process_registry_updates.py

```diff
@@ -184,58 +208,136 @@ def test_ejection_past_churn_limit(spec, state):
+
+    # up to churn limit validators get activated for future epoch from the queue
+    for validator_index in activation_indices[:churn_limit]:
+        validator = state.validators[validator_index]
+        assert validator.activation_eligibility_epoch != spec.FAR_FUTURE_EPOCH
+        assert validator.activation_epoch != spec.FAR_FUTURE_EPOCH
+        assert not spec.is_active_validator(validator, spec.get_current_epoch(state))
+        assert spec.is_active_validator(
+            validator,
+            spec.compute_activation_exit_epoch(spec.get_current_epoch(state))
+        )
+
+    # any remaining validators do not exit the activation queue
+    for validator_index in activation_indices[churn_limit:]:
+        validator = state.validators[validator_index]
+        assert validator.activation_eligibility_epoch != spec.FAR_FUTURE_EPOCH
+        assert validator.activation_epoch == spec.FAR_FUTURE_EPOCH
+
+    # all ejection balance validators ejected for a future epoch
+    for i, validator_index in enumerate(ejection_indices):
+        validator = state.validators[validator_index]
+        assert validator.exit_epoch != spec.FAR_FUTURE_EPOCH
+        assert spec.is_active_validator(validator, spec.get_current_epoch(state))
+        queue_offset = i // churn_limit
+        assert not spec.is_active_validator(
+            validator,
+            spec.compute_activation_exit_epoch(spec.get_current_epoch(state)) + queue_offset
+        )
 
-    # validator activated for future epoch
-    validator = state.validators[activation_index]
-    assert validator.activation_eligibility_epoch != spec.FAR_FUTURE_EPOCH
-    assert validator.activation_epoch != spec.FAR_FUTURE_EPOCH
-    assert not spec.is_active_validator(validator, spec.get_current_epoch(state))
-    assert spec.is_active_validator(
-        validator,
-        spec.compute_activation_exit_epoch(spec.get_current_epoch(state))
-    )
 
-    # validator ejected for future epoch
-    validator = state.validators[ejection_index]
-    assert validator.exit_epoch != spec.FAR_FUTURE_EPOCH
-    assert spec.is_active_validator(validator, spec.get_current_epoch(state))
-    assert not spec.is_active_validator(
-        validator,
-        spec.compute_activation_exit_epoch(spec.get_current_epoch(state))
-    )
+@with_all_phases
+@spec_state_test
+def test_activation_queue_activation_and_ejection__1(spec, state):
+    yield from run_test_activation_queue_activation_and_ejection(spec, state, 1)
+
+
+@with_all_phases
+@spec_state_test
+def test_activation_queue_activation_and_ejection__churn_limit(spec, state):
+    num_validators_per_status = spec.get_validator_churn_limit(state)
+    yield from run_test_activation_queue_activation_and_ejection(spec, state, num_validators_per_status)
+
+
+@with_all_phases
+@spec_state_test
+def test_activation_queue_activation_and_ejection__exceed_churn_limit(spec, state):
+    num_validators_per_status = spec.get_validator_churn_limit(state) + 1
```

Comment:
_may_ be worth `assert`ing that the churn limit is at the floor value in these tests that are not configured to test the case where the limit is higher than the min

Reply:
yes will do